debezium-core/src/main/java/io/debezium/jdbc/JdbcValueConverters.java

```diff
@@ -474,7 +483,7 @@ protected Object convertTimestampToEpochNanos(Column column, Field fieldDefn, Ob
      */
     protected Object convertTimestampToEpochMillisAsDate(Column column, Field fieldDefn, Object data) {
         // epoch is the fallback value
-        return convertValue(column, fieldDefn, data, 0L, (r) -> {
+        return convertValue(column, fieldDefn, data, connectDateTypesUsed() ? new java.util.Date(0) : 0L, (r) -> {
```

Comment:
I don't think this check is needed here, it has been done by the caller before. This method always produces `Date`, also see its name.